pkg/apis/yunikorn.apache.org/update-codegen.sh

```diff
@@ -0,0 +1,31 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+#limitations under the License.
+#
+
+# generate the code with:
+# --output-base    because this script should also be able to run inside the vendor dir of
+#                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
+#                  instead of the $GOPATH directly. For normal projects this can be dropped.
+GO111MODULE=off
+bash k8s.io/code-generator/generate-groups.sh "all" \
+  github.com/apache/incubator-yunikorn-k8shim/pkg/client github.com/apache/incubator-yunikorn-k8shim/pkg/apis \
+  "yunikorn.apache.org:v1alpha1" \
+  #--output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
+  #--go-header-file  ./pkg/apis/apache.org/boilerplate.go.txt
```

Comment:
is this the script to run the code gen? Can you pls add some documents about how to run this?
what's the reason to comment out these 2 lines? For the header file, if this is commented out, how the license headers were added to those generated files?

Reply:
About the license headers, code generator will read the `boilerplate.go.txt` which in `k8s.io/code-generator/hack` first. Base on my test, the `--go-header-file` won't work.
The `--output-base` part, base on the description "because this script should also be able to run inside the vendor dir of k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir instead of the $GOPATH directly. For normal projects this can be dropped." I don't think we need it.

Reply:
> About the license headers, code generator will read the boilerplate.go.txt which in k8s.io/code-generator/hack first. Base on my test, the --go-header-file won't work.

do you mean that specifying a customized file e.g /path/to/asf.header doesn't work with this command? Then how you add those license headers?